crates/noirc_evaluator/src/ssa_refactor/ir/instruction.rs

```diff
@@ -915,12 +917,18 @@ pub(crate) enum SimplifyResult {
     /// a function such as a tuple
     SimplifiedToMultiple(Vec<ValueId>),
 
-    /// Replace this function with an simpler but equivalent function.
-    SimplifiedToInstruction(Instruction),
-
     /// Remove the instruction, it is unnecessary
     Remove,
 
-    /// Instruction could not be simplified
-    None,
+    /// Instruction cannot be simplified to a known value.
+    ///  
+    /// The instruction may be able to be replaced with simpler but equivalent instruction.
+    /// If so it will be included here.
+    SimplifiedToInstruction(Option<Instruction>),
```

Comment:
I don't think we should say we've simplified to a certain instruction if we haven't actually done that and are actually returning None. I think the previous approach read better.

I think we can revert to the previous approach and add a helper method:
```rs
impl SimplifyResult {
    fn instruction(self) -> Option<Instruction> {
        match self {
            SimplifiedToInstruction(instruction) => Some(instruction),
            _ => None,
        }
    }
}
```
Then the callsite can be matching `result @ SimplifiedToInstruction(_) | None => { ... let i = result.instruction().unwrap_or(default)  }`.

Since you said changing each nested call to `simplify` would be annoying, another option is to have a top-level wrapper for simplify which checks if the helper call returned None and if so maps that to `SimplifyResult::Instruction` (renamed since if we didn't do any simplifications it'd be odd to say we simplified it to that instruction).

Reply:
Agreed, your first solution is much cleaner so I've switched to this. I'd like to avoid the wrapper as it's an extra layer of indirection.